modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java

```diff
@@ -513,6 +514,60 @@ public void testUpdateObjectRelationship() throws Exception {
 			objectRelationship6.getObjectFieldId2());
 
 		Assert.assertFalse(objectField2.isRequired());
+
+		ObjectRelationship objectRelationship7 =
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectDefinition2.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap("Able"), StringUtil.randomId(),
+				true, ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		Assert.assertEquals(
+			LocalizedMapUtil.getLocalizedMap("Able"),
+			objectRelationship7.getLabelMap());
+
+		String liferayMode = SystemProperties.get("liferay.mode");
+
+		SystemProperties.clear("liferay.mode");
+
+		try {
+			objectRelationship7 =
+				_objectRelationshipLocalService.updateObjectRelationship(
+					objectRelationship7.getObjectRelationshipId(), 0,
+					ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE,
+					false, LocalizedMapUtil.getLocalizedMap("Baker"));
+		}
+		finally {
+			SystemProperties.set("liferay.mode", liferayMode);
+		}
+
+		Assert.assertEquals(
+			LocalizedMapUtil.getLocalizedMap("Baker"),
+			objectRelationship7.getLabelMap());
+		Assert.assertEquals(
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			objectRelationship7.getDeletionType());
+
+		ObjectRelationship objectRelationship8 =
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectDefinition2.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap("Able"), StringUtil.randomId(),
+				true, ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		objectRelationship8 =
+			_objectRelationshipLocalService.updateObjectRelationship(
+				objectRelationship8.getObjectRelationshipId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE, false,
+				LocalizedMapUtil.getLocalizedMap("Able"));
+
+		Assert.assertEquals(
+			ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE,
+			objectRelationship8.getDeletionType());
```

Comment:
Hey Sá, 
I had an idea on how to organize this test scenario. What do you think?

```
		ObjectRelationship systemObjectRelationship =
			_objectRelationshipLocalService.addObjectRelationship(
				TestPropsValues.getUserId(),
				_objectDefinition1.getObjectDefinitionId(),
				_objectDefinition2.getObjectDefinitionId(), 0,
				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
				LocalizedMapUtil.getLocalizedMap("Able"), StringUtil.randomId(),
				true, ObjectRelationshipConstants.TYPE_MANY_TO_MANY);

		systemObjectRelationship =
			_objectRelationshipLocalService.updateObjectRelationship(
				systemObjectRelationship.getObjectRelationshipId(), 0,
				ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE, false,
				LocalizedMapUtil.getLocalizedMap("Able"));

		Assert.assertEquals(
			ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE,
			systemObjectRelationship.getDeletionType());

		// Not allowed bundles can only edit the label

		String liferayMode = SystemProperties.get("liferay.mode");

		SystemProperties.clear("liferay.mode");

		try {
			systemObjectRelationship =
				_objectRelationshipLocalService.updateObjectRelationship(
					systemObjectRelationship.getObjectRelationshipId(), 0,
					ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
					false, LocalizedMapUtil.getLocalizedMap("Baker"));
		}
		finally {
			SystemProperties.set("liferay.mode", liferayMode);
		}

		Assert.assertEquals(
			LocalizedMapUtil.getLocalizedMap("Baker"),
			objectRelationship7.getLabelMap());
		Assert.assertEquals(
			ObjectRelationshipConstants.DELETION_TYPE_DISASSOCIATE,
			objectRelationship7.getDeletionType());
```

Reply:
I think that would be easier to understand if the test scenario would be:

1. Add a system object relationship
2. Update a system object relationship by an allowed bundle. Showing that we can edit the system object relationship.
3. Update a system object relationship bot a not allowed bundle. Showing the limitation.